proxy/http/HttpSM.cc

```diff
@@ -7793,9 +7784,9 @@ HttpSM::redirect_request(const char *arg_redirect_url, const int arg_redirect_le
 
   bool noPortInHost = HttpConfig::m_master.redirection_host_no_port;
 
-  bool isRedirectUrlOriginForm = clientUrl.m_url_impl->m_len_scheme <= 0 && clientUrl.m_url_impl->m_len_user <= 0 &&
-                                 clientUrl.m_url_impl->m_len_password <= 0 && clientUrl.m_url_impl->m_len_host <= 0 &&
-                                 clientUrl.m_url_impl->m_len_port <= 0;
+  bool isRedirectUrlOriginForm = !clientUrl.m_url_impl->m_len_scheme && !clientUrl.m_url_impl->m_len_user &&
```

Comment:
Is this really equivalent? `-1` is `true` (and `!` makes it `false`), right?

Reply:
It's equivalent because all these members are unsigned, and can not be negative. THat's why CPP check complains, that we're checking an unsigned to be negative (which can't happen of course).

Reply:
I see.